lib/apiservers/engine/backends/system.go

```diff
@@ -240,7 +241,8 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 }
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
-	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", ProductName())
+	defer trace.End(trace.Begin(""))
+	return RegistryService.Auth(authConfig, dockerversion.DockerUserAgent(ctx))
```

Comment:
This is a simple approach, but as @fdawg4l says it pulls in a lot of docker/docker. As best I can tell the docker/docker/registry/... is mainly to abstract between v1 and v2. Given we don't support v2 can we use docker/distribution/registry/client directly?